CommonIncludes/ve_avx512.hpp

```diff
@@ -0,0 +1,1941 @@
+				// fallthrough
+			case 6:
+				dest[e.value + 5] = values[5];
+				// fallthrough
+			case 5:
+				dest[e.value + 4] = values[4];
+				// fallthrough
+			case 4:
+				dest[e.value + 3] = values[3];
+				// fallthrough
+			case 3:
+				dest[e.value + 2] = values[2];
+				// fallthrough
+			case 2:
+				dest[e.value + 1] = values[1];
+				// fallthrough
+			case 1:
+				dest[e.value + 0] = values[0];
+				// fallthrough
+			case 0:
+				break;
+		}
+	}
+
+	template<typename T>
+	RELEASE_INLINE void store(contiguous_tags<T> e, dcon::bitfield_type* dest, vbitfield_type values) {
+		memcpy(dest + e.value / uint32_t(8), &values.v , sizeof(uint16_t));
+	}
+	template<typename T>
+	RELEASE_INLINE void store(unaligned_contiguous_tags<T> e, dcon::bitfield_type* dest, vbitfield_type values) {
+		{
+			uint16_t old_value;
+			std::memcpy(&old_value, dest + e.value / uint32_t(8), sizeof(uint16_t));
+			auto mask = 0x0FFFF << (e.value & 0x07);
+			auto adju_value = values.v << (e.value & 0x07);
+			auto old_masked_out = (old_value & ~mask);
+			auto combined = uint16_t(old_masked_out | adju_value);
+			memcpy(dest + e.value / uint32_t(8), &combined, sizeof(uint16_t));
+		}
+		{
+			if((e.value & 0x07) != 0) {
+				auto old_value = dest[2 + (e.value / uint32_t(8))].v;
+				auto mask = 0x0FF >> (8 - (e.value & 0x07));
+				auto adju_value = values.v >> (16 - (e.value & 0x07));
+				dest[2 + e.value / uint32_t(8)].v = uint8_t((old_value & ~mask) | (adju_value & mask));
+			}
+		}
+	}
+	template<typename T>
+	RELEASE_INLINE void store(partial_contiguous_tags<T> e, dcon::bitfield_type* dest, vbitfield_type values) {
+		{
+			uint16_t old_value;
+			memcpy(&old_value, dest + e.value / uint32_t(8), sizeof(uint16_t));
+			auto mask = (0x0FFFF >> (16 - e.subcount)) << (e.value & 0x07);
+			auto adju_value = values.v << (e.value & 0x07);
+			auto combined = uint16_t((old_value & ~mask) | (adju_value & mask));
+			memcpy(dest + e.value / uint32_t(8), &combined, sizeof(uint16_t));
+		}
+		{
+			if((e.value & 0x07) != 0) {
+				auto old_value = dest[2 + e.value / uint32_t(8)].v;
+				auto mask = (0x0FF >> (16 - e.subcount)) >> (8 - (e.value & 0x07));
+				auto adju_value = values.v >> (16 - (e.value & 0x07));
+				dest[2 + e.value / uint32_t(8)].v = uint8_t((old_value & ~mask) | (adju_value & mask));
+			}
+		}
+	}
+
+	template<typename T>
+	RELEASE_INLINE void store(contiguous_tags<T> e, dcon::bitfield_type* dest, mask_vector values) {
+		store(e, dest, compress_mask(values));
+	}
+	template<typename T>
+	RELEASE_INLINE void store(unaligned_contiguous_tags<T> e, dcon::bitfield_type* dest, mask_vector values) {
+		store(e, dest, compress_mask(values));
+	}
+	template<typename T>
+	RELEASE_INLINE void store(partial_contiguous_tags<T> e, dcon::bitfield_type* dest, mask_vector values) {
+		store(e, dest, compress_mask(values));
+	}
+
+	template<typename U>
+	RELEASE_INLINE void store(tagged_vector<U> indices, float* dest, fp_vector values) {
```

Comment:
for these sorts of functions you might consider doing conflict detection and then scatter if there is no conflict, falling back on the loop otherwise